File: tests/fixtures/python/inheritance.py
"""Python class inheritance examples."""

class User:
    """Base user class."""

    def __init__(self, name: str):
        self.name = name

class Auditable:
    """Auditable mixin."""

    def audit(self) -> None:
        pass

class AdminUser(User, Auditable):
    """Admin user with multiple inheritance."""

    def __init__(self, name: str, role: str):
        super().__init__(name)
        self.role = role

    def get_role(self) -> str:
        return self.role

File: tests/fixtures/java/inheritance.java
package com.example;

public class AdminUser extends User implements Service {
    private String role;

    public AdminUser(String name, int age, String role) {
        super(name, age);
        this.role = role;
    }

    public String getRole() {
        return role;
    }

    @Override
    public void start() {
        // implementation
    }

    @Override
    public void stop() {
        // implementation
    }

    @Override
    public boolean isRunning() {
        return true;
    }
}
